src/ol/style/expressions.js

```diff
@@ -590,6 +592,52 @@ Operators['!'] = {
     return `(!${expressionToGlsl(context, args[0])})`;
   },
 };
+Operators['&&'] = {
+  getReturnType: function (args) {
+    return ValueTypes.BOOLEAN;
+  },
+  toGlsl: function (context, args) {
+    assertArgsCount(args, 2);
+
+    // find common type
+    let type = ValueTypes.ANY;
+    for (let i = 0; i < args.length; i++) {
+      type = type & getValueType(args[i]);
+    }
+    if (type === 0) {
+      throw new Error(
+        `All arguments should be of compatible type, got ${JSON.stringify(
+          args
+        )} instead`
+      );
+    }
+
+    return `(${expressionToGlsl(context, args[0], type)} && ${expressionToGlsl(
+      context,
+      args[1],
+      type
+    )})`;
+  },
+};
+Operators['and'] = {
+  getReturnType: function (args) {
+    return ValueTypes.BOOLEAN;
+  },
+  toGlsl: function (context, args) {
+    assertArgsCount(args, 2);
+    for (let i = 0; i < args.length; i++) {
+      assertBoolean(args[i]);
+    }
+
+    return `(${expressionToGlsl(context, args[0])} && ${expressionToGlsl(
+      context,
+      args[1]
+    )})`;
+  },
+};
+
+Operators['any'] = getEqualOperator('||');
```

Comment:
This should use the same code as the above operator, instead of the equality ones (which can only take two arguments).